### PR TITLE
pythonPackages.typical: init at 2.7.9

### PR DESCRIPTION
**Pull request by kfollesdal**
###### Motivation for this change
Add python package [typical](https://python-typical.org/) to nixpkgs.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by mweinelt — CHANGES_REQUESTED** [review]

**Comment by kfollesdal** [comment]
Thanks @mweinelt for good feedback. I did not get the test for `future-typing` to work. But have added pythonImportsCheck.

**Comment by kfollesdal** [comment]
Result of `nixpkgs-review pr 141515` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python38Packages.future-typing</li>
    <li>python38Packages.typical</li>
    <li>python39Packages.future-typing</li>
    <li>python39Packages.typical</li>
  </ul>
</details>

File: pkgs/development/python-modules/future-typing/default.nix [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  pname = "future-typing";
+  version = "0.4.1";
+
+  src = fetchPypi {
+    pname = "future_typing";
+    inherit version;
+    sha256 = "65fdc5034a95db212790fee5e977fb0a2df8deb60dccf3bac17d6d2b1a9bbacd";
+  };
+
+  doCheck = false; # No tests in pypi source
```
[the remark this answers is not in the document]
Comment (SuperSandro2000) [reply pkgs/development/python-modules/future-typing/default.nix:18]: Then please update the comment.

[the remark this answers is not in the document]
Comment (kfollesdal) [reply pkgs/development/python-modules/future-typing/default.nix]: @SuperSandro2000, I have now update comment.

File: pkgs/development/python-modules/typical/default.nix [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  format = "pyproject";
+
+  src = fetchFromGitHub {
+    owner = "seandstewart";
+    repo = "typical";
+    rev = "v${version}";
+    sha256 = "sha256-ilD8yP1iMPAESsHfQCaY33HjBtrzQ2Px1ylBFcpfIuU=";
+  };
+
+  nativeBuildInputs = [ poetry ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/typical/default.nix:30]: Can we use poetry-core?

Comment (kfollesdal) [reply pkgs/development/python-modules/typical/default.nix:30]: > Can we use poetry-core?

Build fails with `poetry-core`

```
ERROR: Exception:
Traceback (most recent call last):
  File "/nix/store/5w0rmiv65bg1hrsx3wv6p284bnd3z7bx-python3.9-pip-21.1.3/lib/python3.9/site-packages/pip/_internal/cli/base_command.py", line 180, in _main
    status = self.run(options, args)
  File "/nix/store/5w0rmiv65bg1hrsx3wv6p284bnd3z7bx-python3.9-pip-21.1.3/lib/python3.9/site-packages/pip/_internal/cli/req_command.py", line 205, in wrapper
    return func(self, options, args)
  File "/nix/store/5w0rmiv65bg1hrsx3wv6p284bnd3z7bx-python3.9-pip-21.1.3/lib/python3.9/site-packages/pip/_internal/commands/wheel.py", line 142, in run
    requirement_set = resolver.resolve(
  File "/nix/store/5w0rmiv65bg1hrsx3wv6p284bnd3z7bx-python3.9-pip-21.1.3/lib/python3.9/site-packages/pip/_internal/resolution/resolvelib/resolver.py", line 103, in resolve
    r = self.factory.make_requirement_from_install_req(
  File "/nix/store/5w0rmiv65bg1hrsx3wv6p284bnd3z7bx-python3.9-pip-21.1.3/lib/python3.9/site-packages/pip/_internal/resolution/resolvelib/factory.py", line 429, in make_requirement_from_install_req
    cand = self._make_candidate_from_link(
  File "/nix/store/5w0rmiv65bg1hrsx3wv6p284bnd3z7bx-python3.9-pip-21.1.3/lib/python3.9/site-packages/pip/_internal/resolution/resolvelib/factory.py", line 200, in _make_candidate_from_link
    self._link_candidate_cache[link] = LinkCandidate(
  File "/nix/store/5w0rmiv65bg1hrsx3wv6p284bnd3z7bx-python3.9-pip-21.1.3/lib/python3.9/site-packages/pip/_internal/resolution/resolvelib/candidates.py", line 306, in __init__
    super().__init__(
  File "/nix/store/5w0rmiv65bg1hrsx3wv6p284bnd3z7bx-python3.9-pip-21.1.3/lib/python3.9/site-packages/pip/_internal/resolution/resolvelib/candidates.py", line 151, in __init__
    self.dist = self._prepare()
  File "/nix/store/5w0rmiv65bg1hrsx3wv6p284bnd3z7bx-python3.9-pip-21.1.3/lib/python3.9/site-packages/pip/_internal/resolution/resolvelib/candidates.py", line 234, in _prepare
    dist = self._prepare_distribution()
  File "/nix/store/5w0rmiv65bg1hrsx3wv6p284bnd3z7bx-python3.9-pip-21.1.3/lib/python3.9/site-packages/pip/_internal/resolution/resolvelib/candidates.py", line 317, in _prepare_distribution
    return self._factory.preparer.prepare_linked_requirement(
  File "/nix/store/5w0rmiv65bg1hrsx3wv6p284bnd3z7bx-python3.9-pip-21.1.3/lib/python3.9/site-packages/pip/_internal/operations/prepare.py", line 508, in prepare_linked_requirement
    return self._prepare_linked_requirement(req, parallel_builds)
  File "/nix/store/5w0rmiv65bg1hrsx3wv6p284bnd3z7bx-python3.9-pip-21.1.3/lib/python3.9/site-packages/pip/_internal/operations/prepare.py", line 570, in _prepare_linked_requirement
    dist = _get_prepared_distribution(
  File "/nix/store/5w0rmiv65bg1hrsx3wv6p284bnd3z7bx-python3.9-pip-21.1.3/lib/python3.9/site-packages/pip/_internal/operations/prepare.py", line 60, in _get_prepared_distribution
    abstract_dist.prepare_distribution_metadata(finder, build_isolation)
  File "/nix/store/5w0rmiv65bg1hrsx3wv6p284bnd3z7bx-python3.9-pip-21.1.3/lib/python3.9/site-packages/pip/_internal/distributions/sdist.py", line 36, in prepare_distribution_metadata
    self.req.prepare_metadata()
  File "/nix/store/5w0rmiv65bg1hrsx3wv6p284bnd3z7bx-python3.9-pip-21.1.3/lib/python3.9/site-packages/pip/_internal/req/req_install.py", line 549, in prepare_metadata
    self.metadata_directory = self._generate_metadata()
  File "/nix/store/5w0rmiv65bg1hrsx3wv6p284bnd3z7bx-python3.9-pip-21.1.3/lib/python3.9/site-packages/pip/_internal/req/req_install.py", line 534, in _generate_metadata
    return generate_metadata(
  File "/nix/store/5w0rmiv65bg1hrsx3wv6p284bnd3z7bx-python3.9-pip-21.1.3/lib/python3.9/site-packages/pip/_internal/operations/build/metadata.py", line 31, in generate_metadata
    distinfo_dir = backend.prepare_metadata_for_build_wheel(
  File "/nix/store/5w0rmiv65bg1hrsx3wv6p284bnd3z7bx-python3.9-pip-21.1.3/lib/python3.9/site-packages/pip/_vendor/pep517/wrappers.py", line 184, in prepare_metadata_for_build_wheel
    return self._call_hook('prepare_metadata_for_build_wheel', {
  File "/nix/store/5w0rmiv65bg1hrsx3wv6p284bnd3z7bx-python3.9-pip-21.1.3/lib/python3.9/site-packages/pip/_vendor/pep517/wrappers.py", line 275, in _call_hook
    raise BackendUnavailable(data.get('traceback', ''))
pip._vendor.pep517.wrappers.BackendUnavailable: Traceback (most recent call last):
  File "/nix/store/5w0rmiv65bg1hrsx3wv6p284bnd3z7bx-python3.9-pip-21.1.3/lib/python3.9/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 86, in _build_backend
    obj = import_module(mod_path)
  File "/nix/store/97w52ckcjnfiz89h3lh7zf1kysgfm2s8-python3-3.9.6/lib/python3.9/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 972, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
  File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 984, in _find_and_load_unlocked
ModuleNotFoundError: No module named 'poetry.masonry'

```

Comment (SuperSandro2000) [reply pkgs/development/python-modules/typical/default.nix:30]: If you know how to change the code to poetry-core we could make a patch, if not that is also okay.

**Comment by SuperSandro2000** [comment]
Please squash the commits together that we have one commit per package.

**Comment by kfollesdal** [comment]
> Please squash the commits together that we have one commit per package.

@SuperSandro2000, I have now:
* Made a patch so we use poetry-core. 
* Updated to version 2.7.9 for typical
* Squased commits

**Comment by kfollesdal** [comment]
Result of `nixpkgs-review pr 141515` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python38Packages.future-typing</li>
    <li>python38Packages.typical</li>
    <li>python39Packages.future-typing</li>
    <li>python39Packages.typical</li>
  </ul>
</details>

**Pull request merged**